### PR TITLE
fix: Address pagination loop in `meltano-cloud logs print` command

### PR DESCRIPTION
**Pull request by edgarrmondragon**
See https://github.com/meltano/infra/issues/857

### CODE REVIEW

File: src/cloud-cli/meltano/cloud/cli/logs.py [hunk 1]
```diff
@@ -84,11 +84,14 @@ async def get_logs(
                 response = await self._get_logs_page(execution_id, page_token)
                 yield response
 
-                pagination = response.get("pagination")
-                page_token = pagination.get("next_page_token") if pagination else None
-                if not page_token:
+                pagination = response.get("pagination") or {}
+                new_page_token = pagination.get("next_page_token")
+
+                if not new_page_token or new_page_token == page_token:
```
Comment (WillDaSilva) [inline src/cloud-cli/meltano/cloud/cli/logs.py:90]:
```suggestion
                if new_page_token == page_token:
```

**Review by WillDaSilva — APPROVED** [review]
@edgarrmondragon Does this same bug exist with the other places in the Cloud CLI that use pagination?

File: src/cloud-cli/meltano/cloud/cli/logs.py (hunk 1 above)
Comment (edgarrmondragon) [reply src/cloud-cli/meltano/cloud/cli/logs.py:90]: > @edgarrmondragon Does this same bug exist with the other places in the Cloud CLI that use pagination?

@WillDaSilva No, so far only the [`get_log_events`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/logs/client/get_log_events.html) endpoint works like this. DynamoDB pagination works as expected.

**Comment by edgarrmondragon** [comment]
> @edgarrmondragon Does this same bug exist with the other places in the Cloud CLI that use pagination?

@WillDaSilva No, so far only the [`get_log_events`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/logs/client/get_log_events.html) endpoint works like this. DynamoDB pagination works as expected.

**Pull request merged**